service/matching/matchingEngine.go

```diff
@@ -319,7 +327,10 @@ pollLoop:
 		}
 
 		taskList := newTaskListID(domainID, taskListName, persistence.TaskListTypeActivity)
-		tCtx, err := e.getTask(ctx, taskList)
+		// Add frontend generated pollerID to context so tasklistMgr can support cancellation of
+		// long-poll when frontend calls CancelOutstandingPoll API
+		pollerCtx := context.WithValue(ctx, pollerIDKey, pollerID)
```

Comment:
would it possible that pollerID is empty (not supplied by frontend from older version or whatever reason)? Do we need to skip adding the empty pollerID to ctx if it is the case?

Reply:
Good point.  I have added protection in tasklistmgr.gettask to protect against empty pollerID.